content/2. Ondersteunende-informatie/4. Techniek/Git/3. Branches.md

```diff
@@ -0,0 +1,134 @@
+
+In een repository voor een codeer project is het gebruikelijk om in ieder geval een `main` en een `dev`/`development` branch te hebben. Bij een project kan dan elke nieuwe feature een eigen branch krijgen als voorbeeld.
+
+## Branches toevoegen
+Binnen een repository waar jij de rechten hebt om veranderingen te maken aan de repository, kan jij nieuwe branches aanmaken. Een nieuwe branch moet altijd gebaseerd worden op een andere branch (vaak de `main` of `development` branch).
+
+### Lokale en remote branches
+Branches kunnen local of remote zijn. 
+Lokale branches zijn, zoals de naam al zegt, op jouw machine aanwezig, deze zijn (nog) niet zichtbaar voor andere.
+
+[Remote branches](https://git-scm.com/book/en/v2/Git-Branching-Remote-Branches) werken net iets anders. In Git verwijst een remote branch naar een branch die bestaat op een externe repository in plaats van in jouw lokale repository. Remote branches fungeren als aanwijzers naar de status van branches in de externe repository. Je kunt deze branches lokaal bijhouden om op de hoogte te blijven van wijzigingen van andere bijdragers.
+
+Een remote branch kan je identificeren aan het feit dat het in de meeste gevallen `origin/[BRANCH_NAAM]` zal heten. `origin` Is de standaard instelling, deze kan zelf een andere naam krijgen door het clone commando iets anders uit te voeren
+```
+git clone -o booyah
+```
+- Hiermee wordt `origin` vervangen met `booyah`.
+
+
+## Branch protection rules
+Branches kunnen ook beschermd worden tegen bepaalde acties. In de workshop wordt gewerkt met de `classic branch protection rule`, hierbij komen de volgende opties beschikbaar
+![[branch_rule_set.png]]
+
+- `Require a pull request before merging`
+	- Op deze branch kan alleen een verandering plaatsvinden via een pull request. Er kan niet op deze branch direct gecommit.
+- `Require status checks to pass before merging`
+	- Voordat een merge doorgevoerd kan worden moet iets eerst voldoen aan een set eisen. Hieronder een paar voorbeelden:
+		- **Geautomatiseerde Tests**
+			- Zorg ervoor dat alle eenheids-, integratie- en end-to-end tests succesvol worden uitgevoerd.
+		- **Codekwaliteitschecks**
+			- Statistische analysetools kunnen controleren op code stijl, potentiële bugs en andere kwaliteitsmetrics.
+		- **Bouwsucces**
+			- De code moet zonder fouten worden gecompileerd en gebouwd.
+		- **Beveiligingsscans**
+			- Geautomatiseerde tools kunnen controleren op bekende kwetsbaarheden in afhankelijkheden of code.
+		- **Handmatige Reviews**
+			- Sommige workflows vereisen dat een of meer teamleden de wijzigingen beoordelen en goedkeuren.
+- `Require conversation resolution before merging`
+	- Bij een pull request moeten alle discussies zijn opgelost voordat de pull request doorgevoerd kan worden.
```

Comment:
Als PR verder wordt uitgelegd is het handig om te behandelen wat conversations / discussies precies zijn